src/lib/provable/foreign-field.ts

```diff
@@ -180,7 +180,7 @@ class ForeignField {
     Gadgets.ForeignField.assertAlmostReduced(
       xs.map((x) => x.value),
       this.modulus,
-      { skipMrc: true }
```

Comment:
This is quite a deep change that might impact more than just ECDSA, what do others think about going one level higher and applying the multi range checks in the modified `check` methods of ECDSA? So, for example here

```diff
  static check(signature: EcdsaSignature) {
+   // add mrc on signature.r and signature.s
    // more efficient than the automatic check, which would do this for each scalar separately
    this.Curve.Scalar.assertAlmostReduced(signature.r, signature.s);
  }
  ```

Reply:
Yes agree, this is the wrong place

Reply:
The MRC check is done on each foreign field element automatically already, so it will always be a waste of constraints - _except_ inside `check()` methods

Reply:
>The MRC check is done on each foreign field element automatically already, so it will always be a waste of constraints - except inside check() methods

Ah, TIL! Thanks for the input everyone, made those changes!